src/client_side.cc

```diff
@@ -3854,7 +3852,7 @@ ConnStateData::borrowPinnedConnection(HttpRequest *request, const AccessLogEntry
     if (pinning.port != request->url.port())
         throw pinningError(ERR_CANNOT_FORWARD); // or generalize ERR_CONFLICT_HOST
 
-    if (pinning.peer && !cbdataReferenceValid(pinning.peer))
+    if (pinning.peer())
         throw pinningError(ERR_ZERO_SIZE_OBJECT);
```

Comment:
Ouch. This does not work -- we do not want to throw whenever there is a peer. I am not sure how to fix this bug though.

Reply:
I am resolving this change request because there is now a specific fix suggestion in another change request.